lib/model/CModelTools.cc

```diff
@@ -169,7 +164,7 @@ CModelTools::TDouble2Vec CModelTools::CFuzzyDeduplicate::quantize(TDouble2Vec va
 }
 
 core_t::TTime CModelTools::CFuzzyDeduplicate::quantize(core_t::TTime time) const {
-    return maths::CIntegerTools::floor(time, m_TimeEps);
+    return m_TimeEps > 0 ? maths::CIntegerTools::floor(time, m_TimeEps) : time;
```

Comment:
Is there no need for a similar check in `quantize(TDouble2Vec ` where `m_ValueEps` has not be initialised with a call to `computeEpsilons`? That is `m_ValueEps` has length > 0.

Reply:
Yes good catch, the loop should have been changed to be over `m_ValueEps.size()` as part of the changes to make this de-duplicate identical values even when `computeEpsilons` hasn't been called.

Reply:
Odd though as that [unit test](https://github.com/tveasey/ml-cpp-1/blob/081b7a4bcd96e07b4f5d062f8f9bd26eae896582/lib/model/unittest/CModelToolsTest.cc#L95) hits that code and I assume it passes

Reply:
I know, it is surprising. Even more surprisingly is it worked as expected on a run of population analysis on nginx from #57. Strictly the behaviour is "undefined" so it can/does work magically.

In fact, I think the reason it often works is that the type is a small vector, which uses static memory for the first two elements. I suspect that these can easily be <= 0. I tried running valgrind on the unit test and it did indeed show up `conditional jump or move depends on uninitialised value`, so the fact that the `m_ValueEps[0]` turns out to often be less than or equal to zero is chance, it is just seemingly highly likely to occur.